src/database/queries.ts

```diff
@@ -171,54 +171,30 @@ export const getVoucher = async (serialNumber: number): Promise<Voucher> => {
 /**
  * Query the `vouchers` collection and check if a serialNumber is valid.
  */
-export const serialNumberIsValid = async (
+export const getMaxVoucherValue = async (
   serialNumber: number,
-): Promise<boolean> => {
+): Promise<number> => {
   try {
     const docId = serialNumber.toString();
     const docRef = doc(db, 'vouchers', docId);
     // check that serialNumber exists
     const voucherRange = await getVoucherRange(serialNumber);
     if (voucherRange === null) {
-      return false;
+      return 0;
```

Comment:
I think in general we should avoid using dirty return values as a way of indicating that the function has failed, because this is not necessarily intuitive and requires extensive documentation to note. Instead, I'd recommend either throwing an error here and catching it when you use `getMaxVoucherValue`, or you can see how error handling was done in `createVoucher` as an example.

Reply:
++